app/Http/Controllers/ReservationController.php

```diff
@@ -0,0 +1,506 @@
+            }
+
+            self::notifyOnVerifiableReservation($newGroup->firstReservation());
+
+            return redirect()->route(
+                'reservations.show',
+                $newGroup->firstReservation()
+            )->with(
+                'message',
+                $newGroup->verified ? __('general.successful_modification') : __('reservations.verifiers_notified')
+            );
+        } else {
+            // we do not save it yet!
+            $newReservation = new Reservation();
+            $newReservation->reservable_item_id = $item->id;
+            $newReservation->user_id = user()->id;
+            $newReservation->title = $validatedData['title'] ?? null;
+            $newReservation->note = $validatedData['note'];
+            $newReservation->reserved_from = $validatedData['reserved_from'];
+            $newReservation->reserved_until =
+                $item->isWashingMachine()
+                ? Carbon::make($validatedData['reserved_from'])->addHour()
+                : Carbon::make($validatedData['reserved_until']);
+
+
+            $newReservation->verified = user()->can('autoVerify', $item);
+
+            $conflictingReservation = self::hasConflict($newReservation);
+            if ($conflictingReservation) {
+                return redirect()->back()->with(
+                    'error',
+                    __('reservations.already_exists') . ' ' .
+                    "{$conflictingReservation->reserved_from},
+                    {$conflictingReservation->reserved_until}"
+                );
+            }
+
+            // and finally:
+            $newReservation->save();
+            if ($item->isWashingMachine()) {
+                return redirect()->route('reservations.items.index', ['type' => ReservableItemType::WASHING_MACHINE])
+                    ->with('message', __('general.successful_modification'));
+            } else {
+                self::notifyOnVerifiableReservation($newReservation);
+                return redirect()->route('reservations.items.show', $item)
+                    ->with(
+                        'message',
+                        $newReservation->verified ? __('general.successful_modification') : __('reservations.verifiers_notified')
+                    );
+            }
+        }
+    }
+
+    /**
+     * Returns a form for creating a reservation.
+     */
+    public function edit(Reservation $reservation)
+    {
+        // this also makes some other checks
+        $this->authorize('modify', $reservation);
+
+        return view('reservations.edit', [
+            'reservation' => $reservation
+        ]);
+    }
+
+    public const EDIT_THIS_ONLY = 'edit_this_only';
+    public const EDIT_ALL_AFTER = 'edit_all_after';
+    public const EDIT_ALL = 'edit_all';
+
+    /**
+     * Updates a reservation with an edited version.
+     */
+    public function update(Reservation $reservation, Request $request)
```

Comment:
Same, split this into multiple functions. If makes sense, you could create an interface and two more classes for handling washing machines and for rooms.